build_tools/bazel_to_cmake/bazel_to_cmake_targets.py

```diff
@@ -233,37 +233,33 @@ def convert_target(target):
   if target.startswith("//llvm-external-projects/iree-dialects"):
     return _convert_iree_dialects_target(target)
 
+  if target.startswith("@"):
+    raise KeyError(f"No conversion found for target '{target}'")
+
   # IREE root paths map to package names based on explicit rules.
   # If changing these, make the corresponding change in iree_macros.cmake
   # (iree_package_ns function).
 
   # Map //compiler/src/iree/(.*) -> iree::\1 (i.e. iree::compiler::\1)
   m = re.match("^//compiler/src/iree/(.+)", target)
   if m:
-    return ["iree::" + _convert_bazel_path(m.group(1))]
+    return ["iree::" + _convert_to_cmake_path(m.group(1))]
 
   # Map //runtime/src/iree/(.*) -> iree::\1
   m = re.match("^//runtime/src/iree/(.+)", target)
   if m:
-    return ["iree::" + _convert_bazel_path(m.group(1))]
-
-  # Map //runtime/bindings/(.*) -> iree::bindings::\1
-  m = re.match("^//runtime/bindings/(.+)", target)
-  if m:
-    return ["iree::bindings::" + _convert_bazel_path(m.group(1))]
-
-  # Map //samples/(.*) -> iree::samples::\1
-  m = re.match("^//samples[/|:](.+)", target)
-  if m:
-    return ["iree::samples::" + _convert_bazel_path(m.group(1))]
+    return ["iree::" + _convert_to_cmake_path(m.group(1))]
```

Comment:
What about a more general rule of throwing away everything up to a directory called src?

```py
target = "//" + target.rsplit("/src/", maxsplit=1)[1]
```

Reply:
I did group the compiler and runtime rules at first, but opted to keep them explicit. I could go either way.

Reply:
I'd rather have something that's a general rule than a collection of special cases

Reply:
Notably, with the change to "defs", my proposed implementation would no longer work, but we could look for "/src/iree" instead, or actually just look for "/iree/". "Every package is named relative to the last 'iree' directory in its directory path" seems like a nice general rule. We've got some other 'iree' directories. Not sure what the impact of all of those would be:

```shell
$ find . -path ./third_party -prune -o -path ./.git -prune -o -type d -name iree -print
./iree
./docs/website/overrides/.icons/iree
./integrations/tensorflow/python_projects/iree_tf/iree
./integrations/tensorflow/python_projects/iree_tflite/iree
./integrations/tensorflow/python_projects/iree_xla/iree
./integrations/tensorflow/iree-dialects/python/iree
./runtime/src/iree
./runtime/bindings/python/iree
./compiler/src/iree
./compiler/src/iree/compiler/API/python/iree
./llvm-external-projects/iree-dialects/python/iree
```

Reply:
I'd prefer explicit over implicit for the small number of special cases we have. I don't have much of an opinion on naming in the python packages, and a bunch of those are using LLVM style / manually written CMake files anyways, so any rules we set up here wouldn't apply to them.

Reply:
I like clear rules for why something is the way it is. This should make things unsurprising. Whether the implementation uses explicit checks or implicit ones is less important to me. I think the rule here is basically that a "src/iree" directory forms a new root, which seems pretty reasonable. Given that, I'd probably implement it that way (`target.rsplit("src/iree/", maxsplit=1)[1]`), but it doesn't matter too much. I do think we should document it though, as opposed to just "we have special regexes here". I do think the defs targets end up looking a bit weird having "src" in their name. I think they have to be located in that particular directory, which is why they're there. A general rule of dropping "src" directories from the path also seems reasonable, which again could just be implemented in practice as special-casing the defs targets.